src/easylogging++.cc

```diff
@@ -105,25 +105,29 @@ const char* ConfigurationTypeHelper::convertToString(ConfigurationType configura
   return "UNKNOWN";
 }
 
+static struct ConfigurationStringToType {
+  const char* upperCaseStr;
+  const char* lowerCaseStr;
+  ConfigurationType configType;
+} configStringToType[] = {
+  { "ENABLED", "enabled", ConfigurationType::Enabled },
+  { "TO_FILE", "to_file", ConfigurationType::ToFile },
+  { "TO_STANDARD_OUTPUT", "to_standard_output", ConfigurationType::ToStandardOutput },
+  { "FORMAT", "format", ConfigurationType::Format },
+  { "FILENAME", "filename", ConfigurationType::Filename },
+  { "MILLISECONDS_WIDTH", "milliseconds_width", ConfigurationType::MillisecondsWidth },
+  { "SUBSECOND_PRECISION", "subsecond_precision", ConfigurationType::MillisecondsWidth },
+  { "PERFORMANCE_TRACKING", "performance_tracking", ConfigurationType::PerformanceTracking },
+  { "MAX_LOG_FILE_SIZE", "max_log_file_size", ConfigurationType::MaxLogFileSize },
+  { "LOG_FLUSH_THRESHOLD", "log_flush_threshold", ConfigurationType::LogFlushThreshold },
+};
```

Comment:
Talking about ‘pedantic’, the trailing comma is explicitly allowed in the C++ Standard :-). See **§8.5 Initializers** in the Standard.

Also check out these links:

- http://stackoverflow.com/questions/24156241/is-there-any-relevance-to-an-extra-in-the-end-of-a-brace-initialization
- http://stackoverflow.com/questions/29099517/why-is-an-extra-comma-not-allowed-in-a-parameter-list-when-it-is-allowed-in-a-br

The latter mentions:

> K&R allows a trailing comma in an initializer at the end of an initializer-list. The Standard has retained this syntax, since it provides flexibility in adding or deleting members from an initializer list, and simplifies machine generation of such lists.

If you know this can cause problems on a specific compiler you target, I can change it. Normally, I regard it good style to always have this trailing comma. It makes adding new items to the list easier.